putter.go

```diff
@@ -65,80 +113,132 @@ type putter struct {
 // The initial request returns an UploadId that we use to identify
 // subsequent PUT requests.
 func newPutter(url url.URL, h http.Header, c *Config, b *Bucket) (*putter, error) {
-	p := new(putter)
-	p.url = url
-	p.c, p.b = new(Config), new(Bucket)
-	*p.c, *p.b = *c, *b
-	p.c.Concurrency = max(c.Concurrency, 1)
-	p.c.NTry = max(c.NTry, 1)
-	p.bufsz = max64(minPartSize, c.PartSize)
+	ctx, cancel := context.WithCancel(context.TODO())
```

Comment:
`context.TODO()` implies work to be done to determine the right context to use, so this should likely be 
```suggestion
	ctx, cancel := context.WithCancel(context.Background())
```

Reply:
I was thinking that it should be possible for the caller to pass a context in here, but I guess that doesn't really fit with the `io.Writer` paradigm. So I've committed your suggestion.

(On the other hand, there should perhaps be a public `Abort()` command to be used to call the whole thing off. But that doesn't need to be part of this PR.)